src/nvim/drawline.c

```diff
@@ -587,12 +587,13 @@ static void draw_statuscol(win_T *wp, winlinevars_T *wlv, linenr_T lnum, int vir
 
   char buf[MAXPATHL];
   // When a buffer's line count has changed, make a best estimate for the full
-  // width of the status column by building with "w_nrwidth_line_count". Add
-  // potentially truncated width and rebuild before drawing anything.
+  // width of the status column by building with the largest possible line number.
+  // Add potentially truncated width and rebuild before drawing anything.
   if (wp->w_statuscol_line_count != wp->w_nrwidth_line_count) {
     wp->w_statuscol_line_count = wp->w_nrwidth_line_count;
     set_vim_var_nr(VV_VIRTNUM, 0);
-    int width = build_statuscol_str(wp, wp->w_nrwidth_line_count, 0, buf, stcp);
+    int maxlnum = MAX(wp->w_nrwidth_line_count, wp->w_height_inner);
+    int width = build_statuscol_str(wp, maxlnum, maxlnum, buf, stcp);
```

Comment:
Why do `lnum` and `relnum` have the same value here? Shouldn't it be like this?
```suggestion
    int width = build_statuscol_str(wp, wp->w_nrwidth_line_count, wp->w_height_inner, buf, stcp);
```

Reply:
I agree that seems logical but it doesn't work for 'relativenumber' + 'number' where the line count will only be substituted for this estimation when `relnum == 0`, like before. Whereas `relnum == 0` does not yield the right width when the window height is larger than the line count.

I think this together with the logic for the %l item always yields the largest possible width that may be used somewhere in the window.